modules/openglqt/src/processors/canvaswithpropertiesprocessor.cpp

```diff
@@ -0,0 +1,202 @@
+    virtual void onProcessorWidgetPositionChange(ProcessorWidgetMetaData* m) override {
+        position(m);
+    }
+    virtual void onProcessorWidgetDimensionChange(ProcessorWidgetMetaData* m) override {
+        dimension(m);
+    }
+    virtual void onProcessorWidgetVisibilityChange(ProcessorWidgetMetaData* m) override {
+        visible(m);
+    }
+    virtual void onProcessorWidgetFullScreenChange(ProcessorWidgetMetaData* m) override {
+        fullScreen(m);
+    }
+    virtual void onProcessorWidgetOnTopChange(ProcessorWidgetMetaData* m) override { onTop(m); }
+
+    std::function<void(ProcessorWidgetMetaData*)> position;
+    std::function<void(ProcessorWidgetMetaData*)> dimension;
+    std::function<void(ProcessorWidgetMetaData*)> visible;
+    std::function<void(ProcessorWidgetMetaData*)> fullScreen;
+    std::function<void(ProcessorWidgetMetaData*)> onTop;
+};
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo CanvasWithPropertiesProcessor::processorInfo_{
+    "org.inviwo.CanvasWithPropertiesProcessor",  // Class identifier
+    "Canvas With Properties",                    // Display name
+    "Data Output",                               // Category
+    CodeState::Experimental,                     // Code state
+    Tags::GL,                                    // Tags
+};
+const ProcessorInfo CanvasWithPropertiesProcessor::getProcessorInfo() const {
+    return processorInfo_;
+}
+
+CanvasWithPropertiesProcessor::CanvasWithPropertiesProcessor()
+    : Processor()
+    , pwObserver_{std::make_unique<PWObserver>()}
+    , inport_{"inport"}
+    , dimensions_{"dimensions",
+                  "Widget Size",
+                  size2_t(756, 512),
+                  std::pair{size2_t{1}, ConstraintBehavior::Immutable},
+                  std::pair{size2_t{10000}, ConstraintBehavior::Ignore},
+                  size2_t(1, 1),
+                  InvalidationLevel::Valid}
+    , position_{"position",
+                "Widget Position",
+                ivec2(128, 128),
+                std::pair{ivec2{-10000}, ConstraintBehavior::Ignore},
+                std::pair{ivec2{10000}, ConstraintBehavior::Ignore},
+                ivec2(1, 1),
+                InvalidationLevel::Valid,
+                PropertySemantics::Text}
+    , visible_{"visible", "Visible", true, InvalidationLevel::Valid}
+    , fullScreen_{"fullScreen", "Full Screen", false, InvalidationLevel::Valid}
+    , onTop_{"onTop", "On Top", true, InvalidationLevel::Valid}
+    , layerType_{"layerType",
+                 "Visible Layer",
+                 {{"color", "Color layer", LayerType::Color},
+                  {"depth", "Depth layer", LayerType::Depth},
+                  {"picking", "Picking layer", LayerType::Picking}},
+                 0}
+    , layerIndex_{"layerIndex", "Color Layer ID", 0, 0, 0}
+    , paths_{"paths", "Paths", "", InvalidationLevel::InvalidOutput, PropertySemantics::Multiline} {
```

Comment:
Might want to specify "Processor/property paths"
Would probably be nice to be able to group properties somehow

Reply:
that will have to be an other PR

Reply:
Processor/property paths is to long to be visible anyway

Reply:
will add the to the help though